csrc/multidevice/communication.h

```diff
@@ -81,6 +79,7 @@ class Communication {
  private:
   // used for printing
   std::string collective_type_;
+  // FIXME: this seems to be redundant with `root_relative_index_`.
```

Comment:
you're right, however, conceptually, "has_root_" is a compile time constant (at the program level, IOW, it is a `constexpr`), whereas `root_relative_index_` is a runtime variable

Reply:
I'll remove this FIXME. `has_root_` will be gone with #2185.